language_service/src/state/tests.rs

```diff
@@ -1590,34 +2041,52 @@ impl Display for DiagnosticUpdate {
     }
 }
 
-fn new_updater(received_errors: &RefCell<Vec<DiagnosticUpdate>>) -> CompilationStateUpdater<'_> {
+fn new_updater<'a>(
+    received_errors: &'a RefCell<Vec<DiagnosticUpdate>>,
+    received_test_cases: &'a RefCell<Vec<TestCallables>>,
+) -> CompilationStateUpdater<'a> {
     let diagnostic_receiver = move |update: DiagnosticUpdate| {
         let mut v = received_errors.borrow_mut();
         v.push(update);
     };
 
+    let test_callable_receiver = move |update: TestCallables| {
+        let mut v = received_test_cases.borrow_mut();
+        v.push(update);
+    };
+
     CompilationStateUpdater::new(
         Rc::new(RefCell::new(CompilationState::default())),
         diagnostic_receiver,
+        test_callable_receiver,
         TestProjectHost {
             fs: TEST_FS.with(Clone::clone),
         },
+        Encoding::Utf16,
```

Comment:
Gotcha again! Because unit tests are run all in Rust, we use utf8 here. (utf16 is used only when sending locations to JavaScript)

(it really doesn't matter for your test cases now, but this can trip us up in the future if we add a test case in the future that uses unicode characters or something)

```suggestion
        Encoding::Utf16,
```